### PR TITLE
Feature/common utils rebase

### PR DESCRIPTION
**Pull request by dsellarsnr**
* No more symlinks, shared code resolved via relative paths
* Eliminate some of the duplicate code - shared date. iterable, searchTree, string now resolved via relative path and path aliases. 
* Add shell script utility to root (more robustness and documentation coming later)

### CODE REVIEW

File: shared/agent/src/agent.ts [hunk 1]
```diff
@@ -1,10 +1,11 @@
 "use strict";
 // keep this as the first import
 // tslint:disable:ordered-imports
-import * as NewRelic from "newrelic";
 
 import * as fs from "fs";
 import * as os from "os";
+
+import * as NewRelic from "newrelic";
```
Comment (bcanzanella) [inline shared/agent/src/agent.ts:8]: can this line stay as the first import? IIRC the NR agent requires that.

File: shared/util/jest.global.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+module.exports = async function (globalConfig, projectConfig) {
+	process.env.NEW_RELIC_ENABLED = "false";
```
Comment (bcanzanella) [inline shared/util/jest.global.js:2]: nice

**Review by bcanzanella — CHANGES_REQUESTED** [review]
1 comment w/r/t imports

File: shared/agent/src/agent.ts (hunk 1 above)
Comment (dsellarsnr) [reply shared/agent/src/agent.ts]: fixed

**Review by bcanzanella — APPROVED** [review]

**Pull request merged**